### DISCUSSION TITLE
Tracing of slow or failed requests

### DISCUSSION THREAD

**Discussion by ennishol** [quarkusio/quarkus#39506] [Q&A]
Hello,
when using tracing one often is interested in traces for a "problematic" request. For example, if the request took more than 1s or the request resulted in a >= 500 response.

Is it possible to configure traces in such a way that the problematic request traces are always sent to the collector and not filtered out by other means?
Thanks!

[a comment by an automated account is not shown]

**Reply by brunobat** [reply]
Yes, if you implement a custom Sampler. 
See: https://quarkus.io/guides/opentelemetry#sampler

I think this should be a pretty common use case. If you want to create an issue about it or even supply a custom, non default, Sampler to Quarkus, we would accept it.

**Reply by ennishol** [reply]
@brunobat  thanks! I've briefly tried to implement simple sampler to see what is possible. And as far as I can tell there is no information about (my) server response time or code. It has to be somehow added to the context, right? Could you please point me into the right direction?

```

    /** Creates a custom sampler for OpenTelemetry */
    @Produces
    @Singleton
    public Sampler sampler() {
        return new MySampler();
    }

    private static class MySampler implements Sampler {
        @Override
        public SamplingResult shouldSample(Context context, String s, String s1, SpanKind spanKind, Attributes attributes, List<LinkData> list) {
            //http info and spans available but no response time or status code
            return new MySamplingResult(attributes);
        }

        @Override
        public String getDescription() {
            return "custom";
        }

        private static class MySamplingResult implements SamplingResult {
            private final Attributes attributes;

            public MySamplingResult(Attributes attributes) {

                this.attributes = attributes;
            }

            @Override
            public SamplingDecision getDecision() {
                return SamplingDecision.RECORD_AND_SAMPLE;
            }

            @Override
            public Attributes getAttributes() {
                return attributes;
            }
        }
    }
```

**Reply by ennishol** [reply]
Opened a [feature request](https://github.com/quarkusio/quarkus/issues/40064)
